python/tidy/servo_tidy/tidy.py

```diff
@@ -304,41 +304,32 @@ def stdout_redirect(where):
 
 
 def check_lock(file_name, contents):
-    def find_reverse_dependencies(dependency, version, content):
-        dependency_prefix = "{} {}".format(dependency, version)
+    def find_reverse_dependencies(name, version, content):
         for package in itertools.chain([content["root"]], content["package"]):
             for dependency in package.get("dependencies", []):
-                if dependency.startswith(dependency_prefix):
+                if dependency.startswith(name) and version in dependency:
                     yield package["name"]
 
     if not file_name.endswith(".lock"):
         raise StopIteration
 
-    # package names to be neglected (as named by cargo)
+    # Package names to be neglected (as named by cargo)
     exceptions = config["ignore"]["packages"]
 
-    # toml.py has a bug(?) that we trip up in [metadata] sections;
-    # see https://github.com/uiri/toml/issues/61
-    # This should only affect a very few lines (that have embedded ?branch=...),
-    # and most of them won't be in the repo
-    try:
-        content = toml.loads(contents)
-    except:
-        print "WARNING!"
-        print "WARNING! toml parsing failed for Cargo.lock, but ignoring..."
-        print "WARNING!"
-        raise StopIteration
+    content = toml.loads(contents)
 
     packages = {}
     for package in content.get("package", []):
-        packages.setdefault(package["name"], []).append(package["version"])
+        packages.setdefault(package["name"], {})
+        packages[package["name"]].setdefault("version", []).append(package["version"])
+        packages[package["name"]].setdefault("source", []).append(package.get("source", ""))
 
-    for (name, versions) in packages.iteritems():
-        if name in exceptions or len(versions) <= 1:
+    for (name, data) in packages.iteritems():
+        if name in exceptions or len(data["version"]) <= 1:
             continue
 
-        highest = max(versions)
-        for version in versions:
+        highest = max(data["version"])
```

Comment:
Could you also add a similar comment here? That, it's for `Packages with same names and different versions`?